src/main/webapp/definitions/alfresco-content-models.yaml

```diff
@@ -0,0 +1,738 @@
+              $ref: '#/definitions/QNameEntry'
+          source:
+            $ref: '#/definitions/QName'
+  QNameEntry:
+    type: object
+    required:
+      - entry
+    properties:
+      entry:
+        $ref: '#/definitions/QName'
+  QName:
+    type: object
+    required:
+      - namespacePrefix
+      - namespaceUri
+      - name
+    properties:
+      namespacePrefix:
+        type: string
+      namespaceUri:
+        type: string
+      name:
+        type: string
+  ModelEntry:
+    type: object
+    required:
+      - entry
+    properties:
+      entry:
+        $ref: '#/definitions/Model'
+  Model:
+    allOf:
+      - $ref: '#/definitions/QName'
+      - properties:
+          version:
+            description: the model version
+            type: string
+          author:
+            description: author of the model
+            type: string
+          description:
+            description: the model description
+            type: string
+  TypeEntry:
+    type: object
+    required:
+      - entry
+    properties:
+      entry:
+        $ref: '#/definitions/Type'
+  Type:
+    allOf:
+      - $ref: '#/definitions/QName'
+      - properties:
+          title:
+            description: the human-readable type title
+            type: string
+          parent:
+            description: parent type defination
+            $ref: '#/definitions/Type'
+          description:
+            description: description of the type
+            type: string
+          archive:
+            description: define if the type is archived on delete
+            type: boolean
+  AspectEntry:
+    type: object
+    required:
+      - entry
+    properties:
+      entry:
+        $ref: '#/definitions/Aspect'
+  Aspect:
+    allOf:
+      - $ref: '#/definitions/QName'
+  ConstraintEntry:
+    type: object
+    required:
+      - entry
+    properties:
+      entry:
+        $ref: '#/definitions/Constraint'
+  Constraint:
+    allOf:
+      - $ref: '#/definitions/QName'
+      - properties:
+          constraint:
+            description: implementation details about the constrait
+            type: object
+            additionalProperties:
+              type: object
+  PropertyPaging:
+    type: object
+    properties:
+      list:
+        type: object
+        properties:
+          pagination:
+            $ref: '#/definitions/Pagination'
+          entries:
+            type: array
+            items:
+              $ref: '#/definitions/PropertyEntry'
+          source:
+            $ref: '#/definitions/Property'
+  PropertyEntry:
+    type: object
+    required:
+      - entry
+    properties:
+      entry:
+        $ref: '#/definitions/Property'
+  Property:
+    type: object
+    required:
+      - mandatory
+    allOf:
+      - $ref: '#/definitions/QName'
+    properties:
+      constraints:
+        description: list of constraints defined in the property
+        type: array
+        items:
+          $ref: '#/definitions/QName'
```

Comment:
I think this should probably be #/definitions/Constraint

Reply:
We are referring to the QName, so the developer can retrieve the ConstraintDefinition QName using /models/{modelId}/constraints/{constraintId}.
It is ok? Or we should return directly a Constraint object?

Reply:
Yes, return a list of Constraint objects. The QName of the constraint would be available within the Constraint object so they will still be able to look it up if they want.

The important thing here is that the constraints are provided inline (if the client used used the include=constraints option) to avoid the client having to make additional calls to the server.